python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,247 @@
+
+        Raises:
+            TableNotFoundError: If a table with the name does not exist
+        """
+
+    @abstractmethod
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
+        """Create a namespace in the catalog.
+
+        Args:
+            namespace: Namespace identifier
+            properties: A string dictionary of properties for the given namespace
+
+        Raises:
+            AlreadyExistsError: If a namespace with the given name already exists
+        """
+
+    @abstractmethod
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        """Drop a namespace.
+
+        Args:
+            namespace: Namespace identifier
+
+        Raises:
+            NamespaceNotFoundError: If a namespace with the given name does not exist
+            NamespaceNotEmptyError: If the namespace is not empty
+        """
+
+    @abstractmethod
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        """List tables under the given namespace in the catalog.
+
+        If namespace not provided, will list all tables in the catalog.
+
+        Args:
+            namespace: Namespace identifier to search.
+
+        Returns:
+            List[Identifier]: list of table identifiers.
+
+        Raises:
+            NamespaceNotFoundError: If a namespace with the given name does not exist
+        """
+
+    @abstractmethod
+    def list_namespaces(self) -> List[Identifier]:
+        """List namespaces from the given namespace. If not given, list top-level namespaces from the catalog.
+
+        Returns:
+            List[Identifier]: a List of namespace identifiers
+        """
+
+    @abstractmethod
+    def load_namespace_properties(self, namespace: Union[str, Identifier]) -> Properties:
+        """Get properties for a namespace.
+
+        Args:
+            namespace: Namespace identifier
+
+        Returns:
+            Properties: Properties for the given namespace
+
+        Raises:
+            NamespaceNotFoundError: If a namespace with the given name does not exist
+        """
+
+    @abstractmethod
+    def update_namespace_properties(
+        self, namespace: Union[str, Identifier], removals: Optional[Set[str]] = None, updates: Optional[Properties] = None
+    ) -> None:
+        """Removes provided property keys and updates properties for a namespace.
+
+        Args:
+            namespace: Namespace identifier
+            removals: Set of property keys that need to be removed. Optional Argument.
+            updates: Properties to be updated for the given namespace. Optional Argument.
+
+        Raises:
+            NamespaceNotFoundError: If a namespace with the given name does not exist
+            ValueError: If removals and updates have overlapping keys.
+        """
+
+    @staticmethod
+    def identifier_to_tuple(identifier: Union[str, Identifier]) -> Identifier:
```

Comment:
Naming things is one of the hardest things in Computer Science, but I find this one a bit awkward since:
```python
Identifier = Tuple[str, ...]
```
So it is actually already a tuple. How about:
```suggestion
    def to_identifier(identifier: Union[str, Identifier]) -> Identifier:
```